inc/endpoints/class-form-endpoint.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Class file for form endpoints.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\REST_API;
+
+/**
+ * Form Endpoint.
+ */
+class Form_Endpoint extends Endpoint {
+	/**
+	 * Form endpoints to register
+	 *
+	 * @var array
+	 */
+	public $form_endpoints = [];
+
+	/**
+	 * Attach to required hooks for form endpoint
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		add_action( 'rest_api_init', [ $this, 'register_rest_routes' ] );
+		add_filter( 'rest_url', [ $this, 'fix_rest_url' ] );
+	}
+
+	/**
+	 * Register the REST API routes.
+	 */
+	public function register_rest_routes() {
+		/**
+		 * Modify the output of the components route.
+		 *
+		 * @param array $form_endpoints {
+		 *     Form endpoint slugs and callback functions.
+		 *
+		 *     @type string $slug The slug for the form endpoint.
+		 *     @type string $callback response callback to use when the endpoint is called.
+		 * }
+		 */
+		$form_endpoints = (array) apply_filters(
+			'wp_irving_form_endpoints',
+			$this->form_endpoints
```

Comment:
There's no reason to store as a class variable since you never set it. E.g. nowhere in the code does `$this->form_endpoints = something` exist. Might as well keep it all local for now.